### PR TITLE
mat: document that Condition errors are returned by Cholesky methods

### LINKED ISSUE #1470: mat: document places where Condition error is returned
We don't do this elsewhere, though we probably should wherever it's returned; we should say that the error type returned is a `Condition` error when non-nil. This should be a separate PR dealing with all places where this happens.

_Originally posted by @kortschak in https://github.com/gonum/gonum/pull/1441#discussion_r494624335_

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

Closes #1470.

### CODE REVIEW

File: mat/cholesky.go [hunk 1]
```diff
@@ -225,6 +225,8 @@ func (c *Cholesky) LogDet() float64 {
 
 // SolveTo finds the matrix X that solves A * X = B where A is represented
 // by the Cholesky decomposition. The result is stored in-place into dst.
+// If the Cholesky docomposition is singular or near-singular a Condition error
```
Comment (vladimir-ch) [inline mat/cholesky.go:228]: decomposition (with e) everywhere.

File: mat/cholesky.go [hunk 2]
```diff
@@ -778,6 +786,7 @@ func (ch *BandCholesky) SolveTo(dst *Dense, b Matrix) error {
 
 // SolveVecTo finds the vector x that solves A * x = b where A is represented by
 // the Cholesky decomposition. The result is stored in-place into dst.
+// If a non-nil error is returned it will be a Condition error.
```
Comment (vladimir-ch) [inline mat/cholesky.go:789]: Why a different wording here?

Comment (kortschak) [reply mat/cholesky.go:789]: Missed it.

File: mat/cholesky.go (hunk 1 above)
Comment (kortschak) [reply mat/cholesky.go:228]: My eyes are getting bad.

**Comment by kortschak** [comment]
Sorry about that. PTAL

**Review by vladimir-ch — APPROVED** [review]

**Pull request merged**